helpers.js

```diff
@@ -28,10 +28,42 @@ window.processForm = processForm;
  * letter.
  */
 export function shiftLetter (letter, shift) {
-  var code = letter.charCodeAt(0) + shift;
+  // First change the letter into a number.
+  var code = letter.charCodeAt(0);
+  var Acode = 'A'.charCodeAt(0);
+  var Zcode = 'Z'.charCodeAt(0);
+  var acode = 'a'.charCodeAt(0);
+  var zcode = 'z'.charCodeAt(0);
+
+  // Only shift if the letter really is a letter.
+  if (code >= Acode && code <= Zcode) {
+    // The letter is a capital letter. Shift it.
+    code += shift;
+    // If code is too big or too small, wrap it so it isn't.
+    code = wrapNumber(code, Acode, Zcode);
```

Comment:
You could simplify this and the conditionals with using the modulo operator:

```suggestion
    code = 'a'.charCodeAt(0) + letter.charCodeAt(0) % 26
```

Reply:
Dad's comment: Javascript % doesn't work as expected for negative shifts, for example -2 % 26 = -2, not 24. So this way still requires additional logic. But we will think about this some more.

Reply:
Hmm, yes, that's inconvenient :/

Do you just want to add a unit test for now and leave the shift function as it is?

Thanks!

Reply:
I think it works now.

Reply:
Nice! :+1: 

Do you need help with the unit test?

Reply:
No, thank you.

Reply:
All done.